### PR TITLE
Update STE UserTriggerDecision and UserModification events

### PR DESCRIPTION
**Pull request by andrewyuq**
1. add multiple fields in UserTriggerDecision event and UserModification event,
2. Update the latency calculation to reflect the latest definition
3. added some initial a/b testing changes for new auto-trigger UX

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererAutoTriggerService.kt [hunk 1]
```diff
@@ -42,6 +42,7 @@ class CodeWhispererAutoTriggerService : CodeWhispererAutoTriggerHandler, Disposa
 
     private var lastInvocationTime: Instant? = null
     private var lastInvocationLineNum: Int? = null
+    var timeAtLastCharTyped: Instant = Instant.now()
```
Comment (Will-ShaoHua) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererAutoTriggerService.kt:45]: private setter?

File: plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererCodeCoverageTrackerTest.kt [hunk 2]
```diff
@@ -465,48 +442,47 @@ internal class CodeWhispererCodeCoverageTrackerTestPython : CodeWhispererCodeCov
     }
 
     @Test
-    fun `test getAcceptedTokensDelta()`() {
-        val tracker = TestCodePercentageTracker(project, TOTAL_SECONDS_IN_MINUTE, CodeWhispererPython.INSTANCE)
+    fun `test getUnmodifiedAcceptedCharsCount()`() {
```
Comment (Will-ShaoHua) [inline plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererCodeCoverageTrackerTest.kt:445]: should move to cwsprUtilTest.kt, not a blocker for this PR

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererUserModificationTracker.kt [hunk 3]
```diff
@@ -152,28 +156,18 @@ class CodeWhispererUserModificationTracker(private val project: Project) : Dispo
...
-                sendModificationTelemetry(acceptedSuggestion, modificationPercentage)
-                // temp remove event sent as further discussion needed for metric calculation
-                // sendUserModificationTelemetryToServiceAPI(acceptedSuggestion, modificationPercentage)
-            } catch (e: Exception) {
-                sendModificationTelemetry(acceptedSuggestion, null)
-                // temp remove event sent as further discussion needed for metric calculation
-                // sendUserModificationTelemetryToServiceAPI(acceptedSuggestion, 1.0)
+            // Will remove this later when we truly don't need toolkit user modification telemetry anymore
+            val document = runReadAction {
+                FileDocumentManager.getInstance().getDocument(file)
```
Comment (Will-ShaoHua) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererUserModificationTracker.kt:163]: we can do this later, for simplicity, we should simply add document as a field of AcceptedSuggestionEntry. https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorManager.kt#L56

**Review by Will-ShaoHua — APPROVED** [review]

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererAutoTriggerService.kt (hunk 1 above)
Comment (andrewyuq) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererAutoTriggerService.kt:45]: yea

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererFeatureConfigService.kt [hunk 4]
```diff
@@ -84,20 +92,24 @@ class CodeWhispererFeatureConfigService {
...
         featureConfigs[name]?.value ?: FEATURE_DEFINITIONS[name]?.value
-            ?: FeatureValue.builder().boolValue(true).build()
+            ?: FeatureValue.builder().boolValue(false).build()
 
     companion object {
         fun getInstance(): CodeWhispererFeatureConfigService = service()
         private const val TEST_FEATURE_NAME = "testFeature"
         private const val DATA_COLLECTION_FEATURE = "IDEProjectContextDataCollection"
         const val CUSTOMIZATION_ARN_OVERRIDE_NAME = "customizationArnOverride"
+        private const val NEW_AUTO_TRIGGER_UX = "newAutoTriggerUX"
```
Comment (Will-ShaoHua) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererFeatureConfigService.kt:108]: nit: we could make these constants sealed data object

Comment (Will-ShaoHua) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererFeatureConfigService.kt:108]: like
```
sealed interface Feature {
     val  id: String
     
     sealed data object Feature1 { id: 'feat1'}

     sealed data object Feature2 { id: 'feat2' }
}
```

Comment (Will-ShaoHua) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererFeatureConfigService.kt:108]: can do this later

Comment (andrewyuq) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererFeatureConfigService.kt:108]: ok

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererUserModificationTracker.kt (hunk 3 above)
Comment (andrewyuq) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererUserModificationTracker.kt:163]: ok

**Review by rli — APPROVED** [review]

**Pull request merged**